taichi/backends/opengl/opengl_api.cpp

```diff
@@ -243,11 +243,19 @@ struct GLSSBO {
   }
 };
 
-void initialize_opengl() {
+bool initialize_opengl(bool noerr) {
   static bool gl_inited = false;
-  if (gl_inited)
-    return;
-  gl_inited = true;
+  static bool gl_supp = false;
+
+
+  if (gl_inited) {
+    if (!gl_supp) {
+      TI_ASSERT_INFO(noerr, "OpenGL not supported");
```

Comment:
I don't understand this. If `noerr == false`, which means errors are allowed, then this assertion will fail?

Reply:
Oh seems I've made a bad name, actually I mean `noerr -> no_raising_error`, and thus `noerr == true` means errors are allowed.
Sorry about the confusing name :) Maybe we want `noerr -> error_tolerance`?